.github/workflows/auto-assign-issues-prs.yml

```diff
@@ -0,0 +1,43 @@
+name: Auto assign pull requests and issues
+on:
+  pull_request:
+    types: [opened, reopened, edited]
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  assign-issue-and-pr-author:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: toshimaru/auto-author-assign@v1.6.2
+      - uses: octokit/graphql-action@v2.x
+        id: get_linked_issue
+        with:
+          query: |
+            query($repo: String!, $owner: String!, $pr_number: Int!) {
+              repository(owner: $owner, name: $repo) {
+                pullRequest(number: $pr_number) {
+                  id
+                  closingIssuesReferences (first: 50) {
+                    edges {
+                      node {
+                        id
+                        body
+                        number
+                        title
+                      }
+                    }
+                  }
+                }
+              }
+            }
```

Comment:
This query pulls from the PR that has triggered this event some details of the `closingIssuesReferences` https://docs.github.com/en/graphql/reference/objects#pullrequest (though you'll need to scroll down a little bit in the field reference

This is then passed (through `steps.get_linked_issue.outputs.data` to the next step